be/src/exec/vectorized/partition/chunks_partitioner.cpp

```diff
@@ -0,0 +1,232 @@
+// This file is licensed under the Elastic License 2.0. Copyright 2021-present, StarRocks Limited.
+
+#include "exec/vectorized/partition/chunks_partitioner.h"
+
+#include "exprs/expr.h"
+#include "exprs/expr_context.h"
+#include "runtime/current_thread.h"
+
+namespace starrocks::vectorized {
+
+// TODO(hcf) move to utils
+#define RETURN_PTYPE_BYTE_SIZE(TYPE, SIZE)                                           \
```

Comment:
I'm afraid this piece of code has been existed in our code base but just not so generalized. Maybe you could do a little refactor and reuse the existing code.

Reply:
Yes, I add todo on it, I'll fix it next pr